### ISSUE TITLE
support for wlan0 MAC as boardid on rpi0w

### ISSUE THREAD

**Issue by dognotdog** [nerves-project/boardid#19]
On a Raspberry Pi Zero W, I'd like to use the `wlan0` MAC address as the board ID.

Initial digging (eg trying `cat /sys/class/net/wlan0/address`) seems to indicate that it does not exist in the filesystem at erlinit time, would there be other ways to get at it?

**Comment by fhunleth** [comment]
I did some searching. Here's what I found.

The Raspberry Pi derives built-in Ethernet and WiFi MAC addresses from the low 3 bytes of the serial number by default. This can be overridden by programming the OTP. See [OTP bits](https://www.raspberrypi.org/documentation/hardware/raspberrypi/otpbits.md) and [industrial](https://www.raspberrypi.org/documentation/hardware/industrial/README.md).

The formula is to start with the RPi OUI (`B8:27:EB`) or RPi 4 OUI (`DC:A6:32`). For the Ethernet MAC, append the low 3 bytes of the serial number. For the built-in WiFi MAC, XOR the low 3 bytes by 0x555555 and append that.

Example:

```
RPi ZeroW serial number 000000000bd42e6d
The low 3 bytes are d42e6d. 
XOR that by fives gives 817B38. 
Therefore, the WiFi MAC is b8:27:eb:81:7b:38.
```
